Update pipenv

oz123 opened the pull request:
to update pipenv I needed also to update the dependecies (added invoke) and bump the version of parver.

oz123 commented:
QA is broken, because it's checking without the previous commit. If the complete branch is complete branch is merged the build should work.

juippis commented:
No, it's broken because you added python3.7 support to pipenv while parver doesn't have it. So required dependency can't be fulfilled. This error would've been detected by repoman, and I thought repoman stopped committing if it. Judging from the message box you seem to be using repoman to commit so it's a bit weird it let it through :P

juippis commented on dev-python/pipenv/pipenv-2018.11.26.ebuild:
```diff
@@ -0,0 +1,42 @@
+	$(python_gen_cond_dep 'dev-python/backports-shutil_get_terminal_size[${PYTHON_USEDEP}]' 'python2_7')
+	"
+DEPEND="${RDEPEND}
+	test? (
```
Same here. You can inherit `${RDEPEND}` inside `test? ( ...` if the tests are ran, but check if they are needed in DEPEND without test USE.

Otherwise looks good to me. @bucefal91 please also review and test this.

oz123 replied:
As suggested @juippis I moved RPDEPEND into the test block.

oz123 commented on dev-python/parver/parver-0.2.1.ebuild:
```diff
@@ -0,0 +1,22 @@
+PYTHON_COMPAT=( python2_7 python3_{5,6} )
+inherit distutils-r1
+
+DESCRIPTION="parver allows parsing and manipulation of PEP 440 version numbers."
```
fixed.

oz123 commented on dev-python/parver/parver-0.2.1.ebuild:
```diff
@@ -0,0 +1,22 @@
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="dev-python/arpeggio[${PYTHON_USEDEP}]
```
I applied the same scheme as in pipenv.

oz123 commented on dev-python/pipenv/pipenv-2018.11.26.ebuild:
```diff
@@ -0,0 +1,42 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
updated.

oz123 commented:
> No, it's broken because you added python3.7 support to pipenv while parver doesn't have it.

Yes indeed. Upstream metadata didn't specify it. But apparently their CI already test python 3.7 and he ebuild runs find (also with tests activated) for python3.7 in gentoo so I now added it.

juippis commented:
```
!!! Fetched file: parver-0.2.1.tar.gz VERIFY FAILED!
!!! Reason: Filesize does not match recorded size
!!! Got:      25717
!!! Expected: 26316
Refetching... File renamed to '/usr/portage/distfiles/parver-0.2.1.tar.gz._checksum_failure_.bf8kilrj'

!!! Couldn't download 'parver-0.2.1.tar.gz'. Aborting.
 * Fetch failed for 'dev-python/parver-0.2.1', Log file:
 *  '/tmp/portage/dev-python/parver-0.2.1/temp/build.log'
```

Could you double-check this? Rest looks good to me, tests run etc. Let's wait for @gentoo/python and @bucefal91 to comment.

oz123 commented:
> ```
> !!! Fetched file: parver-0.2.1.tar.gz VERIFY FAILED!
> !!! Reason: Filesize does not match recorded size
> !!! Got:      25717
> !!! Expected: 26316
> Refetching... File renamed to '/usr/portage/distfiles/parver-0.2.1.tar.gz._checksum_failure_.bf8kilrj'
> 
> !!! Couldn't download 'parver-0.2.1.tar.gz'. Aborting.
>  * Fetch failed for 'dev-python/parver-0.2.1', Log file:
>  *  '/tmp/portage/dev-python/parver-0.2.1/temp/build.log'
> ```
> 
> Could you double-check this? Rest looks good to me, tests run etc. Let's wait for @gentoo/python and @bucefal91 to comment.

Oh, sorry, I forgot to updated the Manifest file. It's now updated.

juippis commented:
Hey, sorry it's taken so long. I'm not sure do I want to commit on behalf of @gentoo/python since the changed versioning of pipenv. I see it follows upstream, but if python project has some other plans... Could you create a bug about bumping it, then link your PR to that bug? To possibly get a note from python project about what they think about it.

oz123 commented:
@juippis I am not sure it's worth the effort. It seems that pipenv is dead.

prometheanfire commented:
pipenv isn't dead, they have an issue open that asked if they are dead and they said they are not.  They apparently were blocked on some lower level things.  Another release is expected soon.

hartwork commented:
Related: https://github.com/pypa/pipenv/issues/4058#issuecomment-565550646

hartwork commented:
> Hey, sorry it's taken so long. I'm not sure do I want to commit on behalf of @gentoo/python since the changed versioning of pipenv. I see it follows upstream, but if python project has some other plans... Could you create a bug about bumping it, then link your PR to that bug? To possibly get a note from python project about what they think about it.

They put `2018.11.26` into PyPI so they cannot go back to 12.x now because people would not get the latest code from `pip install pipenv` anymore. I don't see much value in using `11.10.4_p20181126` downstream. Let's stick to `2018.11.26`.

oz123 commented:
@hartwork thanks for getting back to me on this. I added your suggestions, and removed what needed to be removed.
I would prefer not to take action on the vendor library right now. This is something that the Python team needs to decide. I am not sure that the current status of pipenv justifies the extra work.
I would be happy to see this PR get merged. If pipenv sees a new release in 2020 we can discuss collaboration on this vendor issue.

hartwork commented on dev-python/pipenv/pipenv-2018.11.26.ebuild:
```diff
@@ -0,0 +1,46 @@
+	>dev-python/requests-2.18.0[${PYTHON_USEDEP}]
+	>=dev-python/urllib3-1.21.1[${PYTHON_USEDEP}]
+	$(python_gen_cond_dep 'dev-python/enum34[${PYTHON_USEDEP}]' 'python2_7')
+	$(python_gen_cond_dep 'dev-python/typing[${PYTHON_USEDEP}]' 'python2_7,python3_4')
```
Python 3.4 is gone from Gentoo by now (and `dev-python/typing` only has `python2_7`)

oz123 replied:
Ok. So this is fixed. I remove python3.4.

hartwork commented:
On a side note I find that `9.0.0-r2` already has the vendor lib problem so it's not something new and maybe not something we need to fix right now, agreed.

oz123 commented:
@hartwork 

> What do you think?

Sounds like a plan  and I pushed a set of changes implementing what you suggested.
